pkg/morph/deploy/notary.go

```diff
@@ -869,3 +878,357 @@ func makeUnsignedDesignateCommitteeNotaryTx(roleContract *rolemgmt.Contract, com
+	committeeMultiSigAcc := wallet.NewAccountFromPrivateKey(prm.localAcc.PrivateKey())
+
+	err := committeeMultiSigAcc.ConvertMultisig(committeeMultiSigM, prm.committee)
+	if err != nil {
+		return fmt.Errorf("compose committee multi-signature account: %w", err)
+	}
+
+	committeeMultiSigAccID := committeeMultiSigAcc.ScriptHash()
+	chNotaryRequests := make(chan *result.NotaryRequestEvent, 100) // secure from blocking
+	// cache processed operations: when main transaction from received notary
+	// request is signed and sent by the local account, we receive the request from
+	// the channel again
+	mProcessedMainTxs := make(map[util.Uint256]struct{})
+
+	subID, err := prm.blockchain.ReceiveNotaryRequests(&neorpc.TxFilter{
+		Signer: &committeeMultiSigAccID,
+	}, chNotaryRequests)
+	if err != nil {
+		return fmt.Errorf("subscribe to notary requests from committee: %w", err)
+	}
+
+	go func() {
+		defer func() {
+			err := prm.blockchain.Unsubscribe(subID)
+			if err != nil {
+				prm.logger.Warn("failed to cancel subscription to notary requests", zap.Error(err))
+			}
+		}()
+
+		prm.logger.Info("listening to committee notary requests...")
+
+		for {
+			select {
+			case <-ctx.Done():
+				prm.logger.Info("stop listening to committee notary requests (context is done)", zap.Error(ctx.Err()))
+				return
+			case notaryEvent, ok := <-chNotaryRequests:
+				if !ok {
+					prm.logger.Info("stop listening to committee notary requests (subscription channel closed)")
+					return
+				}
+
+				// for simplicity, requests are handled one-by one. We could process them in parallel
+				// using worker pool, but actions seem to be relatively lightweight
+
+				const expectedSigners = 3 // sender + committee + Notary
+				mainTx := notaryEvent.NotaryRequest.MainTransaction
+				// note: instruction above can throw NPE and it's ok to panic: we confidently
+				// expect that only non-nil pointers will come from the channel (NeoGo
+				// guarantees)
+
+				srcMainTxHash := mainTx.Hash()
+				_, processed := mProcessedMainTxs[srcMainTxHash]
+
+				switch {
+				case processed:
+					prm.logger.Info("main transaction of the notary request has already been processed, skip",
+						zap.Stringer("ID", srcMainTxHash))
+					continue
+				case notaryEvent.Type != mempoolevent.TransactionAdded:
+					prm.logger.Info("unsupported type of the notary request event, skip",
+						zap.Stringer("got", notaryEvent.Type), zap.Stringer("expect", mempoolevent.TransactionAdded))
+					continue
+				case len(mainTx.Signers) != expectedSigners:
+					prm.logger.Info("unsupported number of signers of main transaction from the received notary request, skip",
+						zap.Int("expected", expectedSigners), zap.Int("got", len(mainTx.Signers)))
+					continue
+				case !mainTx.HasSigner(committeeMultiSigAccID):
+					// in theory, there can be any notary requests besides those sent by the current
+					// auto-deploy procedure. However, it's better to log with 'info' severity since
```

Comment:
i dont have strong opinion about severity of these messages now, so `info` fits better when u dont have administrator's feedback. Ok lets make it `warn`, it's better to watch than to miss

Reply:
done

>  almost all logs in that switch are strange

nope, some of them are pretty natural for Notary service

Reply:
> nope, some of them are pretty natural for Notary service

i meant that if it is a startup procedure, i would not expect some other notary TXs. but ok

Reply:
it's an application startup procedure, but may be not first run of the Sidechain, so other requests are pretty normal here

Reply:
all of them suit my needs because that's how i (admin) track what's goin on right now

Reply:
Right now it's you, and after this it may be another person who may wonder, why WARN log is printing, whether it's OK and should he do something about it. I doubd that for an abstract admin most these INFO/WARN logs are useful.

Reply:
#2419